### ISSUE TITLE
Rule proposal: Forbidden property key

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1283]
1. Forbid known `object` / `array` / `bigint` as property key
2. Fordid unsafe `number` as property key

## Fail

```js
// May lead to unexpected result.
const array = []
const object = {}

const a = {
	[array]: 1,
	[object]() {}
}

class A {
	[array]: 1
	[object]() {}
}
```

```js
// `4n` is used as `4` not `'4n'`
const array = [];
array[4n] = '';
// array.length === 5
```

```js
const a = {
  1.00000000000000001: 1
};
```
([`no-loss-of-precision`](https://eslint.org/docs/rules/no-loss-of-precision) should already covered.)

```js
const a = {
  1234567891012345678910: 1
}
```

## Pass

```js
const array = []
const object = {}

const a = {
	[array.join('')]: 1,
	[toString(object)]: 2
}
```

```js
const array = [];
array[Number(4n)] = '';
```

```js
const symbol = Symbol("key")
const a = {
	[symbol]: 1
}
```

This rule need the data type by static analyze.

**Comment by sindresorhus** [comment]
This is now accepted.
